src/platforms/ios/ioscontroller.swift

```diff
@@ -116,52 +135,31 @@ public class IOSControllerImpl : NSObject {
         }
     }
 
-    @objc func connect(dnsServer: String, serverIpv6Gateway: String, serverPublicKey: String, serverIpv4AddrIn: String, serverPort: Int,  excludeLocalNetworks: Bool, allowedIPAddressRanges: Array<VPNIPAddressRange>, reason: Int, gleanDebugTag: String, isSuperDooperFeatureActive: Bool, installationId: String, disconnectOnErrorCallback: @escaping () -> Void, onboardingCompletedCallback: @escaping () -> Void, vpnConfigPermissionResponseCallback: @escaping (Bool) -> Void) {
+    @objc func connect(serverData: VPNServerData, fallbackServer: VPNServerData?, excludeLocalNetworks: Bool, allowedIPAddressRanges: [VPNIPAddressRange], reason: Int, gleanDebugTag: String, isSuperDooperFeatureActive: Bool, installationId: String, disconnectOnErrorCallback: @escaping () -> Void, onboardingCompletedCallback: @escaping () -> Void, vpnConfigPermissionResponseCallback: @escaping (Bool) -> Void) {
         IOSControllerImpl.logger.debug(message: "Connecting")
 
         TunnelManager.withTunnel { tunnel in
             // Let's remove the previous config if it exists.
             (tunnel.protocolConfiguration as? NETunnelProviderProtocol)?.destroyConfigurationReference()
 
-            let keyData = PublicKey(base64Key: serverPublicKey)!
-            let dnsServerIP = IPv4Address(dnsServer)
-            let ipv6GatewayIP = IPv6Address(serverIpv6Gateway)
-
-            var peerConfiguration = PeerConfiguration(publicKey: keyData)
-            peerConfiguration.endpoint = Endpoint(from: serverIpv4AddrIn + ":\(serverPort )")
-            peerConfiguration.allowedIPs = []
-
-            allowedIPAddressRanges.forEach {
-                if (!$0.isIpv6) {
-                    peerConfiguration.allowedIPs.append(IPAddressRange(address: IPv4Address($0.address as String)!, networkPrefixLength: $0.networkPrefixLength))
-                } else {
-                    peerConfiguration.allowedIPs.append(IPAddressRange(address: IPv6Address($0.address as String)!, networkPrefixLength: $0.networkPrefixLength))
-                }
-            }
-
-            var peerConfigurations: [PeerConfiguration] = []
-            peerConfigurations.append(peerConfiguration)
-
-            var interface = InterfaceConfiguration(privateKey: self.privateKey!)
-
-            if let ipv4Address = IPAddressRange(from: deviceIpv4Address!),
-               let ipv6Address = IPAddressRange(from: deviceIpv6Address!) {
-                interface.addresses = [ipv4Address, ipv6Address]
+            let config = createConfig(for: serverData, allowedIPAddressRanges: allowedIPAddressRanges)
```

Comment:
Potentially out-of-scope: why do we provide the allowedIPAddressRanges to iOS if it doesn't use it?

Reply:
We do, in `ioscontroller.swift` around line 334.

Or do you mean "our architecture doesn't use it"? Great question. It is part of the Wireguard config, which is what is happening there. I'd want to dig in to make sure there isn't something surprising happening before we rip it out - feel free to add a spike ticket if you want - I'd be happy to dig in (outside this ticket).

Reply:
Yeah I mean the architecture. We have a bunch of logic around this list, but it is a constant list (0.0.0.0/0, ::) so all of that is kind of unnecessary and even confusing if you are not aware this is unused.

Reply:
I'm not aware, but I also have dug into the wireguard internals to see what (if anything) they do for it.

Reply:
Wireguard uses that to figure out what should go through the tunnel or not. But we dont rely on it for iOS, we use 'excludeLocalNetworks' instead -- https://github.com/mozilla-mobile/mozilla-vpn-client/blob/main/src/controller.cpp#L513-L524.